mcts/src/main/java/org/motechproject/nms/mcts/service/impl/MctsWsImportServiceImpl.java

```diff
@@ -375,7 +414,36 @@ private MctsImportAudit saveImportedAnmAshaData(AnmAshaDataSet anmAshaDataSet, S
             }
         }
         LOGGER.info("{} state, Total: {} Ashas imported, {} Ashas rejected", stateName, saved, rejected);
-        return new MctsImportAudit(referenceDate, MctsUserType.ASHA, stateCode, stateName, saved, rejected, null);
+        return new MctsImportAudit(startReferenceDate, endReferenceDate, MctsUserType.ASHA, stateCode, stateName, saved, rejected, null);
+    }
+
+    private void deleteMctsImportFailRecords(final LocalDate startReferenceDate, final LocalDate endReferenceDate, final MctsUserType mctsUserType, final Long stateId) {
+
+        @SuppressWarnings("unchecked")
+        SqlQueryExecution<Long> queryExecution = new SqlQueryExecution<Long>() {
+
+            @Override
+            public String getSqlQuery() {
+                String query = "DELETE FROM nms_mcts_failures WHERE importDate between :startReferenceDate AND :endReferenceDate AND stateCode = :stateId AND userType = :mctsUserType";
```

Comment:
I would not recommend running straight sql queries unless you have complex joins that force you to do it. Pull this data through MDS queries is pretty simple and you can iterate over the result set and delete them.

Also, you will have to broadcast an event to flush caches across all instance if you are bypassing MDS with a query like above.